src/components/molecules/ChatList/ChatMessage.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC, useCallback } from "react";
+
+import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
+
+import { WithId } from "utils/id";
+import { formatUtcSeconds } from "utils/time";
+import { getLinkFromText } from "utils/getLinkFromText";
+
+import { useUser } from "hooks/useUser";
+
+import { User } from "types/User";
+
+import {
+  PrivateChatMessage,
+  RestrictedChatMessage,
+} from "components/context/ChatContext";
+
+interface ChatMessageProps {
+  message: WithId<RestrictedChatMessage | PrivateChatMessage>;
+  usersById: Record<string, User>;
+  allowDelete: boolean;
+  onAvatarClick: (
+    message: WithId<RestrictedChatMessage | PrivateChatMessage>
+  ) => void;
+  onDeleteClick: (
+    message: WithId<RestrictedChatMessage | PrivateChatMessage>
+  ) => void;
+}
+
+export const ChatMessage: FC<ChatMessageProps> = ({
+  message,
+  usersById,
+  allowDelete,
+  onAvatarClick,
+  onDeleteClick,
+}) => {
+  const { user } = useUser();
+
+  const profileImageSize = 30;
+  const sender = { ...usersById[message.from], id: message.from };
+  const isMe = sender.id === user!.uid;
```

Comment:
TIL!

https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci/42274019#42274019

> A new `!` post-fix expression operator may be used to assert that its operand is non-null and non-undefined in contexts where the type checker is unable to conclude that fact. Specifically, the operation `x!` produces a value of the type of x with null and undefined excluded. Similar to type assertions of the forms `<T>x` and `x as T`, the `!` non-null assertion operator is simply removed in the emitted JavaScript code.

----

@denisdimitrov just to confirm, that is the expected/desired usage here?

(OOS musings: if so, I wonder if that means our types aren't setup ideally for object?)

Reply:
Doesn't seem like it, but TIL also! Going to change it to `user?.uid`